clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -1204,10 +1199,18 @@ private void testRetriable(final CommitRequestManager commitRequestManager,
                                final List<CompletableFuture<Map<TopicPartition, OffsetAndMetadata>>> futures) {
         futures.forEach(f -> assertFalse(f.isDone()));
 
-        // The manager should backoff for 100ms
-        time.sleep(100);
-        commitRequestManager.poll(time.milliseconds());
+        NetworkClientDelegate.PollResult poll = commitRequestManager.poll(time.milliseconds());
+        assertEquals(0, poll.unsentRequests.size());
+        // The manager should backoff before retry
+        time.sleep(retryBackoffMs);
+        poll = commitRequestManager.poll(time.milliseconds());
+        assertEquals(1, poll.unsentRequests.size());
         futures.forEach(f -> assertFalse(f.isDone()));
+        // Sleep util timeout
+        time.sleep(defaultApiTimeoutMs);
+        poll = commitRequestManager.poll(time.milliseconds());
+        assertEquals(0, poll.unsentRequests.size());
+        futures.forEach(f -> assertTrue(f.isCompletedExceptionally()));
```

Comment:
It seems that `future` doesn't complete exceptionally with `TimeoutException` even we sleep for timeout.
I'm investigation this.